tests/positive/Polymorphism2.mjuvix

```diff
@@ -0,0 +1,44 @@
+module Polymorphism2;
```

Comment:
Did you mean to commit this file - or is it incomplete? It's not run as part of the tests and if I run monojuvix manually I get:

```
minijuvix: the given type is not concrete
CallStack (from HasCallStack):
  error, called at src/MiniJuvix/Prelude/Base.hs:242:9 in minijuvix-0.1.2-AfIq75bwxFQB6Y2Nn2uZ0H:MiniJuvix.Prelude.Base
  error, called at src/MiniJuvix/Syntax/MicroJuvix/Language/Extra.hs:82:14 in minijuvix-0.1.2-AfIq75bwxFQB6Y2Nn2uZ0H:MiniJuvix.Syntax.MicroJuvix.Language.Extra
```

Reply:
this file slipped, my bad.
I don't immediately see what causes the error so I'll need to look into it

Reply:
I found the error. It is fixed in ec2809d0c0998fae4d210ad7622f416c964dd964
The problem was that if a polymorphic inductive definition was never used, it had no associated type calls and thus it was (incorrectly) assumed to be concrete.